internal/expr/parser.go

```diff
@@ -435,6 +417,32 @@ func (p *Parser) skipString(s string) bool {
 	return false
 }
 
+// skipLiteralInList advances the parser to the next comma or closing bracket
+// skipping string literals, comments and matching sets of parentheses.
+func (p *Parser) skipLiteralInList() (bool, error) {
+	for p.pos < len(p.input) {
+		if ok, err := p.skipStringLiteral(); err != nil {
+			return false, err
+		} else if ok {
+			continue
+		}
+		if ok, err := p.skipEnclosedParentheses(); err != nil {
+			return false, err
+		} else if ok {
+			continue
+		}
+		if ok := p.skipComment(); ok {
+			continue
+		}
+
+		if p.char == ',' || p.char == ')' {
+			return true, nil
+		}
+		p.advanceChar()
+	}
+	return false, nil
+}
```

Comment:
This looks like a specific case of the parseUntil function we discussed months ago, maybe it will be helpful now.

Reply:
True, I think we should add that if we add another similar function to this in the future. The `advanceToNextExpression` function where you originally suggested using a `parseUntil` function isn't quite such a good fit for it now, though feel free to bring a PR if you think it could be useful once this is in

Reply:
Only problem with that is that I doubt we are going to remember to refactor it if we need to create another one in the future. If we added it now it wouldn't add more than 5 lines (passing in the closure to get ',' or ')' and return the error).

Reply:
I'd rather leave it as it is, this way the name of the function documents exactly what it does.